### PR TITLE
Update existing test with snapshots

### PR DESCRIPTION
**Pull request by BwyattSSC**
Added snapshot test coverage to existing test. Based on per review, will add additional test as needed.

### Adding vite.config.ts
A `vite.config.ts` have been added to the following directories (see below). This change was made so that the `.snap` files can coincide at the same level of each test.
- `sugar/sugar`
- `sugar/tailwind-helpers/vite.config.ts`
- `sugar/tailwind-plugin-buttons`
- `sugar/tailwind-plugin-form`

There is likely a cleaner way of handling this by extending a higher level `vite.config.ts` file but I haven't found anything on this quite yet.

### Adding snapshots

[sugar/sugar/test/index.test.ts](https://github.com/singlestone/sugar/pull/159/files#diff-7376bd5e00dac2f27b3ca493095c266f7f50714287d4aa30258b0cfc3777847f) assertion was change to use `toMatchInlineSnapshot()` in favor of `toEqual()` to align with using the snapshot approach for testing.

Each test in [sugar/sugar/tailwind-helpers/test](https://github.com/singlestone/sugar/tree/feature/add-initial-vitests-snapshots/sugar/tailwind-helpers/test) have been updated to use `toMatchSnapshot()` instead of `toEqual()`. As such, the expected value in each test has been removed along with its assertion. The assertion is now derived from the snapshot image for each respective test. 

As an example, the [sugar/tailwind-helpers/test/color-helpers.test.ts](https://github.com/singlestone/sugar/pull/159/files#diff-4a48da99d0be443a08e24ee59cc279ad2d683f381978a784aef8cc03193f9207) no longer declares the assertion as part of the constants `noAlphaTests` and `alphaTestsCases`. The assertions are effectively maintained now by [sugar/sugar/tailwind-helpers/test/color-helpers.tests.ts.snap](https://github.com/singlestone/sugar/blob/feature/add-initial-vitests-snapshots/sugar/tailwind-helpers/test/color-helpers.test.ts.snap). In the below screenshot , the alphaTestCase for the first array value has been changed from `#031cfc` to `#099cfc`. When the test run after this change via `pnpm test` from the root of the project, the following messaging is received.

image

This will inform the developer that a change has been made and to action on it accordingly (by updating the application code, test and/or snapshots).

This pattern can also be found in the `sugar/tailwind-plugin-buttons` and `sugar/tailwind-plugin-form` test files as well.

## Related Issue

[ISSUE-160](https://github.com/singlestone/sugar/issues/160)

## Motivation and Context

Pending on perspective, this change may not be required but it would solve the potential future problem of tracking changes in the application along with its expected behavior.

## How Has This Been Tested?

- Run `pnpm install` at the root of the directory
- Navigate to a test and modify its test value
- Save the changes
- Verify that the command line results track the changes and mark the test which are obsolete

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [ ] I added a changeset via `yarn changeset` if my changes should be released.

### CODE REVIEW

**Comment by ToppleTheNun** [comment]
> There is likely a cleaner way of handling this by extending a higher level vite.config.ts file but I haven't found anything on this quite yet.
The cleaner way to do this is to make a package like the tsup-config one: https://github.com/singlestone/sugar/blob/main/tools/tsup-config/src/index.ts

**Comment by BwyattSSC** [comment]
Per conversation with @pcmart03 , moving this item to `Ready for Review`

Left `My code follow the code style of this project` unchecked as I wasn't sure what they were.
Left `My change requires a change to the documentation` and `I have updated documentation accordingly` for the time being. What information would we want to include as it pertains to running the test? Where should this information live?
Left `I added a changeset via yarn changeset if my changes should be released` unchecked as well as I don't believe it needed one but not sure.

File: tools/test-utils/package.json [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    "type": "git",
+    "url": "https://github.com/singlestone/sugar.git",
+    "directory": "tools/test-utils"
+  },
+  "homepage": "https://github.com/singlestone/sugar/tree/main/tools/test-utils#readme",
+  "bugs": {
+    "url": "https://github.com/singlestone/sugar/issues"
+  },
+  "author": "SingleStone Consulting (https://www.singlestoneconsulting.com/)",
+  "license": "MIT",
```
Comment (ToppleTheNun) [inline tools/test-utils/package.json:15]: Can you add `"private": true,` underneath the license field?

**Review by ToppleTheNun — CHANGES_REQUESTED** [review]
One small requested change then we should be good.

File: tools/test-utils/package.json (hunk 1 above)
Comment (BwyattSSC) [reply tools/test-utils/package.json:15]: Commit https://github.com/singlestone/sugar/pull/159/commits/07e91866a67634f0c8e085cdbbcf085004dcdf42 includes the requested changes.

**Review by ToppleTheNun — APPROVED** [review]

**Pull request merged**